x/il/keeper/abci.go

```diff
@@ -0,0 +1,14 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// EndBlocker is called at the end of every block
+func (k Keeper) EndBlocker(ctx sdk.Context) {
+	// iterate over each stoploss position
+	//   fetch pair info from oracle
+	//   calculate current_ratio/reference_pair_ratio
+	//   if slippage > max_slipage
+	//       send eth transaction to withdraw lp_shares liquidity for pair_id
+}
```

Comment:
@jackzampolin @zmanian

Reply:
Lets define the oracle expected keeper:

```
type OracleKeeper interface {
  GetUniswapPair(id string) Pair
}
```

[`Pair`](https://github.com/PeggyJV/sommelier/pull/13/files#diff-269b5141f20cac4c1c5e366d7a245a2b8d1832c7868ebbdc1bfa5428e1eee751R28) definition